controllers/component_controller_conditions.go

```diff
@@ -75,3 +75,28 @@ func (r *ComponentReconciler) SetUpdateConditionAndUpdateCR(ctx context.Context,
 		log.Error(err, "Unable to update Component")
 	}
 }
+
+func (r *ComponentReconciler) SetGitOpsGeneratedConditionAndUpdateCR(ctx context.Context, component *appstudiov1alpha1.Component, generateError error) {
+	log := r.Log.WithValues("Component", component.Name)
+
+	if generateError == nil {
+		meta.SetStatusCondition(&component.Status.Conditions, metav1.Condition{
+			Type:    "GitOpsResourcesGenerated",
+			Status:  metav1.ConditionTrue,
+			Reason:  "OK",
+			Message: "GitOps resource generated successfully",
+		})
+	} else {
+		meta.SetStatusCondition(&component.Status.Conditions, metav1.Condition{
+			Type:    "GitOpsResourcesGenerated",
+			Status:  metav1.ConditionFalse,
+			Reason:  "GenerateError",
```

Comment:
can you also update the file `controllers/applicationsnapshotenvironmentbinding_controller_conditions.go` with this `Reason`

Reply:
or if we want to do `OK` and `Error`, update this instead

Reply:
Updated to use GenerateError.